ast/src/lib.rs

```diff
@@ -101,12 +101,20 @@ impl Ast {
     }
 
     /// Serializes the ast into a JSON string.
-    pub fn to_json_string(&self) -> Result<String, serde_json::Error> {
-        serde_json::to_string_pretty(&self.ast)
+    pub fn to_json_string(&self) -> Result<String, AstError> {
+        Ok(serde_json::to_string_pretty(&self.ast)?)
```

Comment:
Is `Ok` here needed?

Reply:
It is! I have to use ? to convert the serde_json::Errror to an AST Error, so the result is removed, and I need to re-wrap it in an Ok after then.